### PR TITLE
Feat/Add Markdown support to RichInput with Edit/Source toggle

### PR DESCRIPTION
**Pull request by HenryHengZJ**
## Summary

  - UI: Add @tiptap/markdown extension to RichInput editor, switching content format from HTML to Markdown
  - Server: Fix TurndownService escaping markdown characters (e.g. ### Heading → \### Heading) in agentflow variable resolution, fixes https://github.com/FlowiseAI/Flowise/issues/5527

## Before
image
image
When verify from Langsmith, you can see the markdowns are escaped by Turndownservice with a backslash:
image

## After
image
image
image
When verify from Langsmith, you can see the markdowns are sent correctly as it is:
image

##  Backward Compatibility

  - Legacy HTML content is auto-detected and rendered correctly via TipTap's native HTML parsing
  - New content is stored as markdown, and naturally migrates old HTML on re-save

##  Test plan

  - [x] Paste markdown with headings (###), lists (-), bold (**) into RichInput — verify no backslash escaping at runtime
  - [x] Test legacy HTML content loads correctly in both inline editor and expanded dialog
  - [x] Toggle between Edit and Source in expanded dialog — verify content syncs correctly
  - [x] Test variable resolution `{{nodeId.data.instance}}` still works with markdown content
  - [x] Test edge cases: HTML with `<strong>, <table>, <a>` tags are still converted properly by TurndownService

### CODE REVIEW

**Comment by HenryHengZJ** [comment]
cc @jocelynlin-wd @j-sanaa can you guys update the same for the agentflow package richeditor

**Comment by HenryHengZJ** [comment]
> cc @jocelynlin-wd @j-sanaa can you guys update the same for the agentflow package richeditor

hold on, there is issue with resolving variables

**Comment by HenryHengZJ** [comment]
This [commit](https://github.com/FlowiseAI/Flowise/pull/6021/commits/ac6340267cc62f36e2bd9449613ee37844325d2c) resolve custom mention issue:

## Previously

When we change to Source, it resolved into specific format:
image

This causes backend not able to catch the double curly brackets, hence unable to get the actual value.

## After:
image

image

This ensure we are able to catch the double curly brackets with regex and replace with the actual value from server.

File: packages/server/src/utils/buildAgentflow.ts [hunk 1]
```diff
@@ -247,10 +247,13 @@ export const resolveVariables = async (
         // If value is not a string, return as is
         if (typeof value !== 'string') return value
 
-        const turndownService = new TurndownService()
-        value = turndownService.turndown(value)
-        // After conversion, replace any escaped underscores with regular underscores
-        value = value.replace(/\\_/g, '_')
+        // Convert legacy HTML content to markdown, preserving any markdown syntax within
+        if (/<[a-z][a-z0-9]*[^>]*>/i.test(value)) {
```
[the remark this answers is not in the document]
Comment (HenryHengZJ) [reply packages/server/src/utils/buildAgentflow.ts:251]: 1 < 2 wouldn't match though (because 2 isn't immediately after < with a letter). But something like <x> in a math or template context would.

However, in practice for this codebase, the false positive risk is very low

File: packages/ui/src/ui-component/dialog/ExpandRichInputDialog.jsx [hunk 2]
```diff
@@ -24,10 +26,16 @@ import { CustomMention } from '@/utils/customMention'
 
 const lowlight = createLowlight(common)
 
+// Detect if content is legacy HTML (from old getHTML() storage) vs markdown
+const isHtmlContent = (content) => {
+    if (!content || typeof content !== 'string') return false
+    return /<(?:p|div|span|h[1-6]|ul|ol|li|br|code|pre|blockquote|table|strong|em)\b/i.test(content)
```
[the remark this answers is not in the document]
Comment (HenryHengZJ) [reply packages/ui/src/ui-component/dialog/ExpandRichInputDialog.jsx:32]: Missing uncommon tags is actually safe, if `isHtmlContent` returns false on HTML content, it gets treated as markdown and passed to `setContent` with contentType: 'markdown'. The Markdown extension has built-in HTML support, it parses embedded HTML in markdown via TipTap's existing HTML parsing. So the content still renders correctly.

File: packages/ui/src/ui-component/input/RichInput.jsx [hunk 3]
```diff
@@ -131,15 +139,30 @@ export const RichInput = ({ inputParam, value, nodes, edges, nodeId, onChange, d
...
+    // Load initial content after editor is ready, detecting HTML vs markdown
+    useEffect(() => {
+        if (editor && value) {
+            if (isHtmlContent(value)) {
+                editor.commands.setContent(value)
+            } else {
+                editor.commands.setContent(value, { contentType: 'markdown' })
+            }
+        }
+    }, [editor]) // eslint-disable-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (HenryHengZJ) [reply packages/ui/src/ui-component/input/RichInput.jsx:164]: If value is in the dependency array, it creates an infinite loop:

value changes → effect runs → setContent(value) → triggers onUpdate → calls onChange(editor.getMarkdown()) → parent updates value prop → effect runs again → loop

The effect is meant to run once when the editor instance is ready, to load the initial content.
